src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,300 @@
+        if feature_tokens is not None:
+            if len(feature_tokens.shape) != 1:
+                raise ValueError(f"Expected `feature_tokens` shape to be (n, ) but found {feature_tokens.shape}")
+        if audio is not None:
+            if len(audio.shape) != 1:
+                raise ValueError(f"Expected `audio` shape to be (n, ) but found {feature_tokens.shape}")
+        n_bars = self.n_bars if n_bars is None else n_bars
+
+        if beatstep[0] > 0.01:
+            warnings.warn(f"Inference Warning : beatstep[0] is not 0 ({beatstep[0]}). all beatstep will be shifted.")
+            beatstep = beatstep - beatstep[0]
+
+        if self.use_mel:
+            batch, ext_beatstep = self.preprocess_mel(
+                audio,
+                beatstep,
+                n_bars=n_bars,
+                padding_value=self.padding_value,
+            )
+        else:
+            raise NotImplementedError("use_mel must be True")
+
+        return batch, ext_beatstep
+
+    def __call__(
+        self,
+        raw_audio: Union[np.ndarray, List[float], List[np.ndarray]],
+        audio_sr: int,
+        steps_per_beat: int = 2,
+        return_tensors: Optional[Union[str, TensorType]] = "pt",
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        Args:
+        Main method to featurize and prepare for the model one sequence. Please note that `Pop2PianoFeatureExtractor`
+        only accepts one raw_audio at a time.
+            raw_audio (`np.ndarray`, `List`):
+                Denotes the raw_audio.
+            audio_sr (`int`):
+                Denotes the Sampling Rate of `raw_audio`.
+            steps_per_beat (`int`, *optional*, defaults to 2):
+                Denotes Steps per beat.
+            return_tensors (`str` or [`~utils.TensorType`], *optional*, defaults to 'pt'):
+                If set, will return tensors instead of list of python integers. Acceptable values are:
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return Numpy `np.ndarray` objects.
+        """
+        warnings.warn("If you are not getting optimal performence, please try audio sampling_rate as 44100.")
+        warnings.warn(
+            "Pop2PianoFeatureExtractor only takes one raw_audio at a time, if you want to extract features from more than a single audio then you might need to call it multiple times."
+        )
+
+        # If it's [np.ndarray]
+        if isinstance(raw_audio, list) and isinstance(raw_audio[0], np.ndarray):
+            raw_audio = raw_audio[0]
+
+        bpm, beat_times, confidence, estimates, essentia_beat_intervals = self.extract_rhythm(raw_audio=raw_audio)
+        beatsteps = self.interpolate_beat_times(beat_times, steps_per_beat, extend=True)
+
+        if self.sampling_rate != audio_sr and self.sampling_rate is not None:
+            # Change audio_sr to self.sampling_rate
+            raw_audio = librosa.core.resample(
```

Comment:
Usually we leave resampling to the user: https://github.com/huggingface/transformers/blob/90e8263d912daf8bdb6c3849d35b6588cf5cc39c/src/transformers/models/wav2vec2/feature_extraction_wav2vec2.py#L171

This way, we don't need an extra `librosa` import. But since we're already requiring two extra python packages, maybe it's fine here to have `librosa` as an additional dependency as well

Reply:
I tried to remove the resampling part but here raw_audio is used 2 times, first time in `extract_rhythm` which takes audio with original sampling_rate and the second time in `single_preprocess` which first upscales/downscales raw_audio to 22050(as I noticed it needs sampling_rate  as 22050, otherwise it was giving bad midi results) and then it is passed to the model. Even if we take raw_audio at 22050 from user, `extract_rhythm` wont work properly, if we take raw_audio at original sampling rate from user then `single_preprocess` wont work as expected! So we need to sample at least once here

Reply:
I find it odd that a rhythm extractor would work worse with 22050 Hz than with 44.1 kHz, since rhythm is primarily produced by low and mid frequencies (except maybe hihats). 

Unless maybe Essentia assumes that the input is always 44.1 kHz, since I don't see any way to pass in the sampling rate. Which would mean all input to this model always *must be* 44.1 kHz (right now the docs make it seem like this is optional).

Ah yes, I see in the docs that it's required to be 44100 Hz. We can't avoid resampling then. In `extract_rhythm` it should raise an error if the passed in sampling rate for the example is not 44100.